### PR TITLE
8322759: Eliminate -Wparentheses warnings in compiler code

### PR DESCRIPTION
**Pull request by kimbarrett**
Please review this change to eliminate some -Wparentheses warnings. This
involved simply adding a few parentheses to make some implicit operator
precedence explicit.

This change addresses non-C2 parts of the compiler component.

Testing: mach5 tier1

Also ran mach5 tier1 with these changes in conjunction enabling -Wparentheses
and other changes needed to make that work.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8322759](https://bugs.openjdk.org/browse/JDK-8322759): Eliminate -Wparentheses warnings in compiler code (**Enhancement** - P4)

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to [b2a4515a](https://git.openjdk.org/jdk/pull/17200/files/b2a4515adb347832b4c9fa0747876f96377389a4)
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17200/head:pull/17200` \
`$ git checkout pull/17200`

Update a local copy of the PR: \
`$ git checkout pull/17200` \
`$ git pull https://git.openjdk.org/jdk.git pull/17200/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17200`

View PR using the GUI difftool: \
`$ git pr show -t 17200`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17200.diff">https://git.openjdk.org/jdk/pull/17200.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17200#issuecomment-1871704791)

### CODE REVIEW

File: src/hotspot/share/compiler/compilerDefinitions.inline.hpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016, 2022, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2016, 2023, Oracle and/or its affiliates. All rights reserved.
```
Comment (vnkozlov) [inline src/hotspot/share/compiler/compilerDefinitions.inline.hpp:2]: 2024

**Review by vnkozlov — APPROVED** [review]
Looks good.

File: src/hotspot/share/compiler/compilerDefinitions.inline.hpp [hunk 2]
```diff
@@ -57,7 +57,7 @@ inline bool CompilerConfig::is_c1_only() {
 }
 
 inline bool CompilerConfig::is_c1_or_interpreter_only_no_jvmci() {
-  assert(is_jvmci_compiler() && is_jvmci() || !is_jvmci_compiler(), "JVMCI compiler implies enabled JVMCI");
+  assert((is_jvmci_compiler() && is_jvmci()) || !is_jvmci_compiler(), "JVMCI compiler implies enabled JVMCI");
```
Comment (shipilev) [inline src/hotspot/share/compiler/compilerDefinitions.inline.hpp:60]: This looks like simply:

```
assert(!is_jvmci_compiler() || is_jvmci(), "JVMCI compiler implies enabled JVMCI");
```

File: src/hotspot/share/compiler/compilerDefinitions.inline.hpp [hunk 3]
```diff
@@ -114,7 +114,7 @@ inline bool CompilerConfig::is_c2_or_jvmci_compiler_only() {
 
 // Tiered is basically C1 & (C2 | JVMCI) minus all the odd cases with restrictions.
 inline bool CompilerConfig::is_tiered() {
-  assert(is_c1_simple_only() && is_c1_only() || !is_c1_simple_only(), "c1 simple mode must imply c1-only mode");
+  assert((is_c1_simple_only() && is_c1_only()) || !is_c1_simple_only(), "c1 simple mode must imply c1-only mode");
```
[the remark this answers is not in the document]
Comment (kimbarrett) [reply src/hotspot/share/compiler/compilerDefinitions.inline.hpp]: Agreed.  Changed accordingly.

**Review by shipilev — APPROVED** [review]
Looks reasonable, thanks!

**Comment by kimbarrett** [comment]
Thanks for reviews @vnkozlov and @shipilev .

**Pull request closed without merging**